aya/src/bpf.rs

```diff
@@ -510,7 +510,7 @@ impl<'a> BpfLoader<'a> {
 
         obj.relocate_maps(
             maps.iter()
-                .map(|(s, data)| (s.as_str(), data.fd, &data.obj)),
+                .map(|(s, data)| (s.as_str(), data.fd().as_fd().as_raw_fd(), &data.obj)),
```

Comment:
why not pass a `BorrowedFd<'_>` here instead? The less references we have to `RawFd`s the easier it'll be gauge that our file descriptors are all valid.

Reply:
IIRC the function we're calling here is available in no_std, so `BorrowedFd` can't appear in its signature.

Reply:
Made that function available only in `std`. Unclear to me how anyone would use such a thing without access to real file descriptors.

Reply:
I think we should undo the aya-obj bit. Relocation puts file descriptors in the instruction stream where they must be cast as integers anyway, so lifetime tracking doesn't buy us much and it makes aya-obj essentially useless on no_std.

Reply:
I can do that. But can you help me understand how someone would use this in no_std given you can't get your hands on an FD? Or is the idea that the user has an FD through some other means?

Reply:
Yes, it's unfortunate that *Fd are defined in `std::os`, in an ideal world some of the `os` definitions should be moved to `core` like it happened for `net`. But in a `no_std` program it's easy to imagine that one could still create fds using lower level linux APIs, and it seems silly that we would preclude them from using aya-obj just for a type alias.

Reply:
I don't think this is equivalent to the `core::net` types since those are just scalars like IP addresses. Anyway, done -- I just defined a local alias for RawFd in this crate in no_std.